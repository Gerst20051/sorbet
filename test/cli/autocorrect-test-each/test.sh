#!/bin/bash

tmp="$(mktemp)"
infile="test/cli/autocorrect-test-each/each.rb"
cp "$infile" "$tmp"

# Run the autocorrect in place, on the temp file
main/sorbet --silence-dev-message -a "$tmp" 2> /dev/null

# cat the file to see the autocorrected output
cat "$tmp"

echo "=========="

# Verify that we made the right autocorrects for `test_each`
main/sorbet --silence-dev-message "$tmp" 2>&1

rm "$tmp"
