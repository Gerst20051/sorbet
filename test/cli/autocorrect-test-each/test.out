# typed: true
extend T::Sig

def it(name, &blk); end
def test_each(val, &blk); end
def foo(x); end

test_each([1, 2]) do |x|
  it "asdf #{x}" do
    z = foo(x)
  end
end

test_each([[1, 2], [3, 4]]) do |(x, y)|
  it "asdf #{x} #{y}" do
    z = foo(x) + foo(y)
  end
end

test_each({1 => 2, 3 => 4}) do |x, y|
  it "asdf #{x} #{y} hash" do
  end
end

test_each({1 => 2, 3 => 4}) do |a|
  it "asdf #{a} hash" do
  end
end

test_each([1,2]) do |x|
  it "asdf #{x} test_each" do
    z = foo(x)
  end
end
==========
No errors! Great job.
